src/systems/variational_smoother_system.C

```diff
@@ -113,90 +114,124 @@ void VariationalSmootherSystem::prepare_for_smoothing()
+          target_elem_inverse_jacobian_dets[elem->type()] =
+              std::vector<Real>(nq_points, 1.0);
+
+          // Elems deriving from Tri
+          if (elem->type() == TRI3 || elem->type() == TRI6)
             {
-              // Build target element: an equilateral triangle
-              Tri3 target_elem;
 
-              // equilateral triangle side length that preserves area of reference
-              // element
+              // The target element will be an equilateral triangle with area equal to
+              // the area of the reference element.
+
+              // First, let's define the nodal locations of the vertices
               const Real sqrt_3 = std::sqrt(3.);
-              const auto ref_volume = target_elem.reference_elem()->volume();
-              const auto s = std::sqrt(4. / sqrt_3 * ref_volume);
+
+              std::vector<Point> equilateral_points{
+                Point(0.,  0.),
+                Point(1.,  0.),
+                Point(0.5, 0.5 * sqrt_3)
+              };
+
+              // Target element
+              const auto target_elem = Elem::build(elem->type());
+
+              // Area of the reference element
+              const auto ref_area = target_elem->reference_elem()->volume();
+
+              switch (elem->type())
+                {
+                case TRI3:
+                  {
+                    // Nothing to do here, vertices already defined in equilateral_points
+                    break;
+                  }
+
+                case TRI6:
+                  {
+                    // Define the midpoint nodes of the equilateral triangle
+                    equilateral_points.emplace_back(0.50, 0.          );
+                    equilateral_points.emplace_back(0.75, 0.25 * sqrt_3);
+                    equilateral_points.emplace_back(0.25, 0.25 * sqrt_3);
+
+                    break;
+                  }
+
+                default:
+                  libmesh_error_msg("Unsupported triangular element!");
+                  break;
+                }
+
+              // Equilateral triangle side length preserving area of the reference element
+              const auto side_length = std::sqrt(4. / sqrt_3 * ref_area);
+
+              std::vector<std::unique_ptr<Node>> owned_nodes;
 
               // Set nodes of target element to form an equilateral triangle
-              Node node_0 = Node(0., 0.);
-              Node node_1 = Node(s, 0.);
-              Node node_2 = Node(0.5 * s, 0.5 * s * sqrt_3);
-              target_elem.set_node(0, &node_0);
-              target_elem.set_node(1, &node_1);
-              target_elem.set_node(2, &node_2);
+              for (const dof_id_type node_id : index_range(equilateral_points))
+                {
+                  // Scale the nodal positions to conserve area
+                  auto node_ptr = std::make_unique<Node>(
+                      equilateral_points[node_id] * side_length, node_id);
```

Comment:
`Node::build()` can be used here, and I'd emplace_back() the new Node directly into `owned_nodes`, catching the reference returned by that to then pass to `target_elem->set_node(...)`. This just saves the step of defining the `node_ptr` temporary variable.